src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -240,14 +240,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PackageDownload Include="Microsoft.Net.Sdk.Compilers.Toolset" Version="[$(NETCoreSdkVersion)]" />
   </ItemGroup>
 
-  <Target Name="_CheckMicrosoftNetSdkCompilersToolsetPackageReference" Condition="'$(MSBuildRuntimeType)' == 'Full'" BeforeTargets="CollectPackageReferences">
+  <Target Name="_CheckMicrosoftNetCompilersToolsetFrameworkPackageReference" Condition="'$(MSBuildRuntimeType)' == 'Full'" BeforeTargets="CollectPackageReferences">
     <!-- Users should not be setting Microsoft.Net.Compilers.Toolset.Framework directly.
          If they do, and BuildWithNetFrameworkHostedCompiler is also set, the former will override the latter.
          This makes it more explicit that that is not supported. -->
     <NETSdkWarning ResourceName="CannotDirectlyReferenceMicrosoftNetCompilersToolsetFramework"
         Condition="'@(PackageReference->AnyHaveMetadataValue('Identity', 'Microsoft.Net.Compilers.Toolset.Framework'))' == 'true'" />
   </Target>
 
+  <Target Name="_CheckMicrosoftNetSdkCompilersToolsetPackageExists" Condition="'$(_NeedToDownloadMicrosoftNetSdkCompilersToolsetPackage)' == 'true'" BeforeTargets="CoreCompile">
+    <!-- If users did not run restore or it failed to download the Microsoft.Net.Sdk.Compilers.Toolset package
+         (but they proceeded to build, e.g., in Visual Studio), display an error with suggestions how to fix the problem. -->
+    <NETSdkError ResourceName="MicrosoftNetSdkCompilersToolsetNotFound"
```

Comment:
This could also be just a warning since technically the build will fall back to the compiler bundled with MSBuild - I don't know if that's a scenario we want to error or warn about. I chose error because that can be always relaxed without breaking users.

Reply:
> I chose error because that can be always relaxed without breaking users unlike the other way around.

I endorse this philosophy.

Reply:
Although I worry about breaking users, perhaps a warning would be better (it could be suppressed). @jaredpar also mentioned this should be a warning right?